src/query-parser.ts

```diff
@@ -1,18 +1,31 @@
 import { Dictionary } from "@navarik/types"
 import { SearchQuery } from "./types"
 
+const castType = (value, type) => {
+  if (!type){return value}
+  if (type.includes('int') || type.includes('long')) {
+    return Number(value)
+  }
+  else if (type.includes('boolean')) {
+    return Boolean(value)
+  }
+  else return value
+}
+
 export class QueryParser {
-  parse(searchParams: Dictionary<any>): SearchQuery {
+  parse(searchParams: Dictionary<any>, schemas: object): SearchQuery {
```

Comment:
Do `schemas` change with every call to `parse()`? if not, then they should be the QueryParser's constructor argument